data-prepper-plugins/mutate-string-processors/src/test/java/com/amazon/dataprepper/plugins/processor/mutatestring/UppercaseStringProcessorTests.java

```diff
@@ -0,0 +1,144 @@
+        when(config.getWithKeys()).thenReturn(Arrays.asList("message", "message2"));
+
+        final UppercaseStringProcessor processor = createObjectUnderTest();
+        final Record<Event> record = getEvent("thisisamessage");
+        record.getData().put("message2", "test2");
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) processor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.get(0).getData().containsKey("message"), is(true));
+        assertThat(editedRecords.get(0).getData().get("message", Object.class), equalTo("THISISAMESSAGE"));
+        assertThat(editedRecords.get(0).getData().containsKey("message2"), is(true));
+        assertThat(editedRecords.get(0).getData().get("message2", Object.class), equalTo("TEST2"));
+    }
+
+    @Test
+    public void testHappyPathMultiMixedUppercaseStringProcessor() {
+        when(config.getWithKeys()).thenReturn(Arrays.asList("message", "message2"));
+
+        final UppercaseStringProcessor processor = createObjectUnderTest();
+        final Record<Event> record = getEvent("thisisamessage");
+        record.getData().put("message2", 3);
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) processor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.get(0).getData().containsKey("message"), is(true));
+        assertThat(editedRecords.get(0).getData().get("message", Object.class), equalTo("THISISAMESSAGE"));
+        assertThat(editedRecords.get(0).getData().containsKey("message2"), is(true));
+        assertThat(editedRecords.get(0).getData().get("message2", Object.class), equalTo(3));
+    }
+
+    @Test
+    public void testValueIsNotStringUppercaseStringProcessor() {
+        final UppercaseStringProcessor processor = createObjectUnderTest();
+        final Record<Event> record = getEvent(3);
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) processor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.get(0).getData().containsKey("message"), is(true));
+        assertThat(editedRecords.get(0).getData().get("message", Object.class), equalTo(3));
+    }
+
+    @Test
+    public void testValueIsNullUppercaseStringProcessor() {
+        final UppercaseStringProcessor processor = createObjectUnderTest();
+        final Record<Event> record = getEvent(null);
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) processor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.get(0).getData().containsKey("message"), is(true));
+        assertThat(editedRecords.get(0).getData().get("message", Object.class), equalTo(null));
+    }
+
+    @Test
+    public void testValueIsObjectUppercaseStringProcessor() {
+        final UppercaseStringProcessor processor = createObjectUnderTest();
+        final TestObject testObject = new TestObject();
+        testObject.a = "MSG";
```

Comment:
nit: consider making this lowercase since this class is testing for Uppercase strings even though this is not related.